db.go

```diff
@@ -660,6 +666,31 @@ func (a *dbAppender) Rollback() error {
 	return g.Wait()
 }
 
+// Delete implements deletion of metrics.
+func (db *DB) Delete(mint, maxt int64, ms ...labels.Matcher) error {
+	db.cmtx.Lock()
+	defer db.cmtx.Unlock()
```

Comment:
I will make this change in later PR and would prefer if this PR went in with the current implementation.